src/main/scheduler/scheduler.c

```diff
@@ -564,36 +614,47 @@ FAST_CODE void scheduler(void)
             taskRequiredTimeCycles += taskGuardCycles;
 
             if (!gyroEnabled || (taskRequiredTimeCycles < schedLoopRemainingCycles)) {
+                uint32_t antipatedEndCycles = nowCycles + taskRequiredTimeCycles;
                 taskExecutionTimeUs += schedulerExecuteTask(selectedTask, currentTimeUs);
                 nowCycles = getCycleCounter();
-                schedLoopRemainingCycles = cmpTimeCycles(nextTargetCycles, nowCycles);
+                int32_t cyclesOverdue = cmpTimeCycles(nowCycles, antipatedEndCycles);
+
+#if defined(USE_LATE_TASK_STATISTICS)
+                if (cyclesOverdue > 0) {
+                    if ((currentTask - tasks) != TASK_SERIAL) {
+                        DEBUG_SET(DEBUG_SCHEDULER_DETERMINISM, 1, currentTask - tasks);
+                        DEBUG_SET(DEBUG_SCHEDULER_DETERMINISM, 2, clockCyclesTo10thMicros(cyclesOverdue));
+                        currentTask->lateCount++;
+                        lateTaskCount++;
+                        lateTaskTotal += cyclesOverdue;
+                    }
+                }
+#endif  // USE_LATE_TASK_STATISTICS
 
-                if (schedLoopRemainingCycles < taskGuardMinCycles) {
+                if (-cyclesOverdue < taskGuardMinCycles) {
```

Comment:
what?

Reply:
Should have been
```
                if ((cyclesOverdue > 0) || (-cyclesOverdue < taskGuardMinCycles)) {
```

We need to increase the guard band is we're overdue, or within time, but by less than taskGuardMinCycles.